### ISSUE TITLE
--skipUserCfg is available for csources-built nim binary, but is ineffective

### ISSUE THREAD

**Issue by kaushalmodi** [nim-lang/Nim#10031]
Hello,

While trying to find a workaround for https://github.com/nim-lang/Nim/issues/10030, I ended up with this issue.

Above issue's workaround is to *not* have the user config.nims in the search path when building using the csources-built basic `nim` binary.

I did `--fullhelp` for the csources built `nim`, and it showed the `--skipUserCfg` option as available. But that option is ineffective .. it doesn't actually skip the user config.nims.

See below (`./bin/nim` in that example is the `nim` binary built from csources) that it still is fetching `/home/kmodi/.config/nim/config.nims`:

```
km²~/downloads/:git/Nim> ./bin/nim c koch --skipUserCfg         12/17 1:44pm
Hint: used config file '/home/kmodi/downloads/git/Nim/config/nim.cfg' [Conf]
Hint: used config file '/home/kmodi/downloads/git/Nim/koch.nim.cfg' [Conf]
Hint: used config file '/home/kmodi/.config/nim/config.nims' [Conf]
lib/system.nim(376, 50) Warning: unknown magic 'Destroy' might crash the compiler [UnknownMagic]
lib/pure/os.nim(705, 14) Error: undeclared identifier: 'Stat'
```

**Comment by kaushalmodi** [comment]
Looks like `./bin/nim --skipUserCfg c koch` is the correct way .. moving `--skipUserCfg` to the very beginning, before `c koch` correctly bypasses the user config.

**Comment by timotheecour** [comment]
so is there a bug here?
or should we update https://github.com/nim-lang/Nim/blob/bb1160b73cc3410ce7a0949f67d2fa7f5b425ab4/build_all.sh#L29
from `echo_run bin/nim c koch` to `echo_run bin/nim --skipUserCfg c koch` ?
/cc @Araq

**Comment by kaushalmodi** [comment]
> from echo_run bin/nim c koch to echo_run bin/nim --skipUserCfg c koch 

That's a workaround. So I don't think `--skipUserCfg` should be baked in there. As I mention in https://github.com/nim-lang/Nim/issues/10030, even if we bypass that issue in the `koch`-building step with this `--skipUserCfg`, the same issue comes up in the `./koch boot -d:release` step too.

And we cannot pass `--skipUserCfg` to `koch`.. So I end up with the workaround I do in https://github.com/kaushalmodi/hello_musl/commit/ed97b4b248fa88097ef6a655219b5ea9722cf738.

**Comment by timotheecour** [comment]
ya that workaround is pretty ugly, especially if it affects every .travis.yml for every nim project

**Comment by Araq** [comment]
I think it's ok for now. Can't backport these fixes to 0.19.2 as they are too risky but 0.19.9 has much better ``config.nims`` support.

**Comment by timotheecour** [comment]
=> https://github.com/nim-lang/Nim/pull/10244
